build.gradle

```diff
@@ -108,8 +104,42 @@ apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
 apply from: 'jmh.gradle'
 
+project(':iceberg-relocate-external-dependencies') {
+  apply plugin: 'com.github.johnrengelman.shadow'
+  
+  tasks.javadoc.enabled = false
+  tasks.javadocJar.enabled = false
+  tasks.assemble.dependsOn tasks.shadowJar
+  tasks.install.dependsOn tasks.shadowJar
+  
+  dependencies {    
+    compile('com.google.guava:guava:28.0-jre') {
+      exclude group: 'com.google.code.findbugs'
+      // may be LGPL - use ALv2 findbugs-annotations instead
+      exclude group: 'com.google.errorprone'
+      exclude group: 'com.google.j2objc'
+    }
+  }
+  
+  shadowJar {
+    configurations = [project.configurations.compile]
+    zip64 true
+
+    // include the LICENSE and NOTICE files for the shaded Jar
+    from(projectDir) {
+      include 'LICENSE'
+      include 'NOTICE'
+    }
+
+    relocate 'com.google.common', 'com.google.common.shaded'
```

Comment:
I don't want to trigger a big set of changes, but just wanted to hear what people think about this. Is `com.google.common.shaded` too generic and can conflict? Should everything we shade be something like `iceberg.[original.package.name]`

Reply:
I think `com.google.common.shaded` is too generic. I would go with `org.apache.iceberg.[original.package.name]` like I originally had and then the classes all line up together in the same top-level folder in the jar file etc. Alternatively one could just do `iceberg` as the prefix but I think it looks a bit strange as it doesn't really follow the "reverse TLD" packaging pattern.

Reply:
If you feel strongly about packaging, then I think it's fine to put the classes under the `org.apache.iceberg.` namespace.